src/docfx/restore/PackageResolver.cs

```diff
@@ -44,9 +44,13 @@ public bool TryResolvePackage(PackagePath package, PackageFetchOptions options,
             }
         }
 
-        public Package ResolveAsPackage(PackagePath package, PackageFetchOptions options)
+        public Package ResolveAsPackage(PackagePath package, PackageFetchOptions options, FileResolver fileResolver)
```

Comment:
consider putting `FileResolver` as a member, if there is circular dependency, use `Lazy<FileResolver>`
```suggestion
        public Package ResolveAsPackage(PackagePath package, PackageFetchOptions options)
```